package.json

```diff
@@ -1,35 +1,58 @@
 {
-  "name": "@govtechsg/document-store-contract",
+  "name": "@govtechsg/document-store",
   "version": "3.0.0",
   "main": "index.js",
   "repository": "git@github.com:Open-Attestation/document-store-contract.git",
   "author": "GovTech",
   "license": "Apache-2.0",
   "scripts": {
+    "build:sol": "truffle build",
+    "build:js": "tsc --emitDeclarationOnly && babel src -d dist --ignore src/**/*.spec.ts,src/**/*.test.ts -x .js,.ts,.tsx --copy-files",
+    "build": "npm run build:sol && npm run build:js",
     "commit": "git-cz",
     "commit:retry": "npm run commit -- --retry",
-    "eslint:fix": "eslint . --ext .js --fix",
-    "eslint": "eslint . --ext .js",
+    "lint:js": "eslint . --ext .js",
+    "lint:js:fix": "eslint . --ext .js --fix",
     "lint:sol": "./node_modules/.bin/solhint contracts/**/*.sol",
     "lint:sol:fix": "./node_modules/.bin/prettier --write contracts/**/*.sol",
-    "lint:fix": "npm run lint:sol:fix && npm run eslint:fix",
-    "lint": "npm run lint:sol && npm run eslint",
-    "test": "truffle test",
+    "lint": "npm run lint:sol && npm run lint:js",
+    "lint:fix": "npm run lint:sol:fix && npm run lint:js:fix",
+    "test:sol": "truffle test",
+    "test:js": "jest --testPathPattern=src",
+    "test": "npm run test:sol && npm run test:js",
     "truffle": "truffle",
-    "benchmark": "truffle test ./benchmark/*"
+    "benchmark": "truffle test ./benchmark/*",
+    "typechain": "typechain --target ethers \"build/contracts/*.json\"",
+    "postinstall": "npm run build:sol && npm run typechain"
```

Comment:
Unfortunately this wont work if someone pull this lib as a dependency, because dev dependencies wont be fetched but postinstall will be run. 

Can follow what I did on open-attestation

Reply:
Should I move typechain and truffle into prod? This will allow it to be built on non unix. ( I think ? )

Reply:
yeah can, dont forget to share with the team also on this and put a note in the readme (the idea is to try to prevent someone to move this back to dev dependencies)